packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -539,6 +543,23 @@ class _MergeableMaterialState extends State<MergeableMaterial> {
   }
 }
 
+class _MergeableMaterialSliceKey extends GlobalKey {
```

Comment:
[answering my own question] No. This would be similar to a GlobalValueKey, which would compare values with == as ValueKey does. However the framework does not provide such a public class because in general it would not be possible to guarantee that all GlobalValueKeys had unique values.